### PR TITLE
Configure Vulture to skip tests/unit_tests/

### LINKED ISSUE #1186: Configure Vulture to skip unit test directory
# Spike

## Objective
Find a way to skip unit tests for vulture checks. Otherwise the code that is not being used anywhere is still passing vulture because unit tests use it.

## Tasks
- [ ] Find a way to skip unit tests for vulture checks and implement it. Also, rename `whitelist` to `vulture_whitelist`. (0.25d) - @shreyamalviya

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do? 

Fixes #1186 

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing? 
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...} 
* [ ] If applicable, add screenshots or log transcripts of the feature working

## Explain Changes

Are the commit messages enough? If not, elaborate.

### CODE REVIEW

File: vulture_whitelist.py [hunk 1]
```diff
@@ -186,3 +186,20 @@
 LOG_DIR_NAME  # unused variable (envs/monkey_zoo/blackbox/log_handlers/test_logs_handler.py:8)
 delete_logs  # unused function (envs/monkey_zoo/blackbox/test_blackbox.py:85)
 MongoQueryJSONEncoder  # unused class (envs/monkey_zoo/blackbox/utils/json_encoder.py:6)
+
+
+# exists in common/, monkey_island/, or infection_monkey/, but is used in tests/unit_tests/ which is skipped by Vulture
```
Comment (VakarisZ) [inline vulture_whitelist.py:191]: So why can't these files be moved to unit tests dir?

File: vulture_whitelist.py [hunk 2]
```diff
@@ -186,3 +186,20 @@
...
+_.instance_name  # unused attribute (monkey/common/cloud/azure/azure_instance.py:35)
+_.instance_name  # unused attribute (monkey/common/cloud/azure/azure_instance.py:64)
```
Comment (VakarisZ) [inline vulture_whitelist.py:193]: AFAIK unused. Probably we need to at least add a new section to the top with a comment of `# Unused code awaiting cleanup` or something.

File: vulture_whitelist.py [hunk 3]
```diff
@@ -186,3 +186,20 @@
...
+SomeDummyPlugin  # unused class (monkey/infection_monkey/utils/plugins/pluginTests/BadImport.py:4)
+BadPluginInit  # unused class (monkey/infection_monkey/utils/plugins/pluginTests/BadInit.py:4)
+BadInit  # unused class (monkey/infection_monkey/utils/plugins/pluginTests/ComboFile.py:4)
+ProperClass  # unused class (monkey/infection_monkey/utils/plugins/pluginTests/ComboFile.py:9)
+PluginWorking  # unused class (monkey/infection_monkey/utils/plugins/pluginTests/PluginWorking.py:4)
```
Comment (VakarisZ) [inline vulture_whitelist.py:198]: All of these look like they belong in unit tests

File: vulture_whitelist.py [hunk 4]
```diff
@@ -186,3 +186,20 @@
...
+environment  # unused variable (monkey/monkey_island/cc/models/monkey.py:59)
+_.environment  # unused attribute (monkey/monkey_island/cc/services/telemetry/processing/system_info_collectors/environment.py:10)
```
Comment (VakarisZ) [inline vulture_whitelist.py:200]: Let's move these to `potentially unused`, but I think island extracts it with a mongo query somewhere and vulture is just not picking it up

File: vulture_whitelist.py [hunk 5]
```diff
@@ -186,3 +186,20 @@
...
+ISLAND  # unused variable (monkey/monkey_island/cc/services/utils/node_states.py:14)
+MONKEY_LINUX_RUNNING  # unused variable (monkey/monkey_island/cc/services/utils/node_states.py:26)
```
Comment (VakarisZ) [inline vulture_whitelist.py:202]: Used in `_is_state_from_keywords`

File: vulture_whitelist.py [hunk 6]
```diff
@@ -186,3 +186,20 @@
...
+RAW_SCOUTSUITE_DATA  # unused variable (monkey/monkey_island/cc/services/zero_trust/test_common/raw_scoutsute_data.py:3)
+save_example_findings  # unused function (monkey/monkey_island/cc/services/zero_trust/zero_trust_report/test_common/example_finding_data.py:8)
```
Comment (VakarisZ) [inline vulture_whitelist.py:204]: Should be in unit tests

File: vulture_whitelist.py [hunk 7]
```diff
@@ -186,3 +186,20 @@
...
+data_for_tests_dir  # unused function (monkey/tests/conftest.py:11)
```
Comment (VakarisZ) [inline vulture_whitelist.py:205]: Used in `monkey/tests/unit_tests/monkey_island/cc/test_encryptor.py`

**Review by VakarisZ — CHANGES_REQUESTED** [review]
Address the comments regarding unused files

**Comment by TheCoreMan** [comment]
If you're already renaming, allow list is better than white list, for example https://bugs.chromium.org/p/chromium/issues/detail?id=981129

**Comment by VakarisZ** [comment]
Good one :D

File: vulture_whitelist.py (hunk 5 above)
Comment (shreyamalviya) [reply vulture_whitelist.py:202]: Moved to the first section

File: vulture_whitelist.py (hunk 3 above)
Comment (shreyamalviya) [reply vulture_whitelist.py:198]: Moved

File: pyproject.toml [hunk 8]
```diff
@@ -23,5 +23,5 @@ addopts = "-v --capture=sys tests"
 norecursedirs = "node_modules dist"
 
 [tool.vulture]
-exclude = ["monkey/monkey_island/cc/ui"]
+exclude = ["monkey/monkey_island/cc/ui/", "monkey/tests/unit_tests/"]
```
Comment (VakarisZ) [inline pyproject.toml:26]: why not skip the whole `monkey/tests`?

File: vulture_allowlist.py [hunk 9]
```diff
@@ -182,7 +184,15 @@
...
 delete_logs  # unused function (envs/monkey_zoo/blackbox/test_blackbox.py:85)
 MongoQueryJSONEncoder  # unused class (envs/monkey_zoo/blackbox/utils/json_encoder.py:6)
+environment  # unused variable (monkey/monkey_island/cc/models/monkey.py:59)
+_.environment  # unused attribute (monkey/monkey_island/cc/services/telemetry/processing/system_info_collectors/environment.py:10)
+_.instance_name  # unused attribute (monkey/common/cloud/azure/azure_instance.py:35)
+_.instance_name  # unused attribute (monkey/common/cloud/azure/azure_instance.py:64)
+
+
+# used in tests/unit_tests/ which is skipped by Vulture
+data_for_tests_dir  # unused function (monkey/tests/conftest.py:11)
```
Comment (VakarisZ) [inline vulture_allowlist.py:198]: Why do we need this? why not skip the whole monkey/tests?

**Review by VakarisZ — CHANGES_REQUESTED** [review]
Answer and ping me

**Review by VakarisZ — APPROVED** [review]

**Pull request merged**